pkg/agent/proxy/proxier.go

```diff
@@ -218,59 +257,82 @@ func (p *proxier) installServices() {
 			addedLoadBalancerIPs = svcInfo.LoadBalancerIPStrings()
 		}
 		if len(deletedLoadBalancerIPs) > 0 || len(addedLoadBalancerIPs) > 0 {
-			needUpdate = true
+			needUpdateService = true
 		}
 
-		if !needUpdate {
+		// If neither the Service nor Endpoints of the Service need to be updated, we skip.
+		if !needUpdateService && !needUpdateEndpoints {
 			continue
 		}
 
-		if err := p.ofClient.InstallEndpointFlows(svcInfo.OFProtocol, endpointUpdateList, p.isIPv6); err != nil {
-			klog.Errorf("Error when installing Endpoints flows: %v", err)
-			continue
-		}
-		err := p.ofClient.InstallServiceGroup(groupID, svcInfo.StickyMaxAgeSeconds() != 0, endpointUpdateList)
-		if err != nil {
-			klog.Errorf("Error when installing Endpoints groups: %v", err)
-			p.endpointInstalledMap[svcPortName] = nil
-			continue
+		op := "Installing"
+		if pSvcInfo != nil {
+			op = "Updating"
 		}
-		// Delete previous flow.
-		if needRemoval {
-			if err := p.ofClient.UninstallServiceFlows(pSvcInfo.ClusterIP(), uint16(pSvcInfo.Port()), pSvcInfo.OFProtocol); err != nil {
-				klog.Errorf("Failed to remove flows of Service %v: %v", svcPortName, err)
+		klog.V(2).Infof("%s Service %s %s", op, svcPortName.Name, svcInfo.String())
```

Comment:
I'd rather have 2 log messages to be honest, I think it's more readable:
```go
		if pSvcInfo != nil {
                          klog.V(2).Infof("Installing...")
		} else {
                          klog.V(2).Infof("Updating...")
                 }
```

Reply:
Updated.